app/Models/Foundation/Summit/Summit.php

```diff
@@ -6868,4 +6875,47 @@ public function removeRegistrationFeedMetadata(SummitRegistrationFeedMetadata $m
         $metadata->clearSummit();
     }
 
+    /**
+     * @return SummitLeadReportSetting[]
+     */
+    public function getLeadReportSettings()
+    {
+        return $this->lead_report_settings;
+    }
+
+    /**
+     * @param SummitLeadReportSetting $lead_report_setting
+     */
+    public function addLeadReportSetting(SummitLeadReportSetting $lead_report_setting): void
+    {
+        $this->lead_report_settings->add($lead_report_setting);
+    }
+
+    /**
+     * @param SummitLeadReportSetting $lead_report_setting
+     * @return void
+     */
+    public function removeLeadReportSetting(SummitLeadReportSetting $lead_report_setting):void
+    {
+        if(!$this->lead_report_settings->contains($lead_report_setting)) return;
+        $this->lead_report_settings->removeElement($lead_report_setting);
+        $lead_report_setting->clearSummit();
+    }
+
+    /**
+     * @param Sponsor|null $sponsor
+     * @return SummitLeadReportSetting|null
+     */
+    public function getLeadReportSettingFor(?Sponsor $sponsor = null):?SummitLeadReportSetting {
+        $criteria = Criteria::create();
```

Comment:
this logic is not correct
please review it
u are assuming that the sponsor has one
and it could be null and on that case u need to return the one set for the summit